soc/arm/nxp_imx/rt/Kconfig.defconfig.series

```diff
@@ -92,6 +92,9 @@ config FLASH_SIZE
 	default $(dt_node_int_prop_int,$(DT_CHOSEN_FLASH_NODE),size,Kb) \
 		if $(DT_FLASH_HAS_SIZE_PROP)
 
+config MEMC
```

Comment:
Setting this will cause us to always pull in/build the FlexSPI memc driver as well. This could be an issue if the FlexSPI is not configured properly for XIP on a customer's board (and it will also increase the image size).

Is there any way we could avoid enabling this unless the FlexRAM bank allocation is different from the fused reset value?

Reply:
> Setting this will cause us to always pull in/build the FlexSPI memc driver as well. 

That sounds like an issue. Maybe the FlexSPI Kconfig should depend on DT 

> Is there any way we could avoid enabling this unless the FlexRAM bank allocation is different from the fused reset value?

That doesn't seem to fix the problem you're pointing out

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/62997

Reply:
Looking at the definition of the `MEMC_MCUX_FLEXSPI` Kconfig, this specific case won't be an issue- the memc driver Kconfig does not default to enabled based on the DT compatible, so it won't be enabled unless a flash (or another memc) driver `selects` it: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/memc/Kconfig.mcux#L27

This is still a broader issue we need to consider though, because if we added a MEMC driver for something like the SEMC then we would not want that driver to be enabled in every build.

> That doesn't seem to fix the problem you're pointing out

No, it doesn't- but it reduces the scope. The fact is that any time you enable a driver class like this by default on an EVK board, you are potentially pulling in multiple drivers (since the EVK will set most peripherals to `status = "okay"`). We've seen this recently with the PXP DMA driver being enabled for DMA API tests. I think we should be careful enabling an entire driver class like this in all builds.

Reply:
> This is still a broader issue we need to consider though, because if we added a MEMC driver for something like the SEMC then we would not want that driver to be enabled in every build.

That's exactly one of the cases I had in mind with #62997

Reply:
> I think we should be careful enabling an entire driver class like this in all builds.

An alternative is to put this in misc, but I mean, memc is already a misc folder that is pretending not to be, the crux of all these issues is the fact that all the Kconfigs are only sourced if one grand Kconfig, `MEMC` is set, which is not a real subsystem and I don't why do we pretend like it is in Zephyr and impose unnecessary constraints and hoops to jump through like this

Reply:
can you clarify what is your feedback for this PR

Reply:
Truthfully, just musing about how to best work around this problem. This PR highlights an issue with the memc "subsystem" that I think we should solve in a separate PR, so I won't try to expand the work here.